### DISCUSSION TITLE
Should we add external dependencies ? like material-ui or ngx-translate ?

### DISCUSSION THREAD

**Discussion by moolsbytheway** [myndpm/open-source#17] [Dynamic Forms]
I say no.
I think we should stay away from this kind of coupling.
we can introduce themes the user can choose from or customise the css classes by himself.

**Comment by matheo** [comment]
Indeed @moolsbytheway 
but those dependencies will be required for specific implementations, not inside the library of course.

All the ui-subpackages will have to install `ng-zorro`, `taiga-ui`, `bootstrap`, etc, but those are not required by `@myndpm/dyn-forms`.

About `i18n`, I know from the Angular Team directly, that runtime language-changes won't be supported, so we address that topic with the given tools, and if we need to change the language of the form in runtime, we will need to add `@ngx-translate` reactive parameters into our application, but not inside the library :slightly_smiling_face: 

If you want to build `DynControls` independent of any framework, you can build them with ease, like removing all the Material stuff from the example [ui-material](https://github.com/myndpm/open-source/tree/master/libs/forms/ui-material/src) subpackage, and leaving some basic styles to make them consistent (and then creating a PR with them to be included as a `ui-native` subpackage hehe). I've built the `ui-material` just as a demostration of the power of the library, it's not mandatory to use it :wink: 

All the unused `ui-subpackages` will be tree-shaked from the Angular bundle so there's no big deal having them in the package IMO.

CSS classes can be added to each node of the form-hierarchy too, but still you will need the native controls to work with, because at this point of the library, we don't have all the UI possibilities supported :slightly_smiling_face:
still, we have a ton of features that are in the roadmap of many dynamic-forms libraries out there isn't it? WDYT?
